File: client/src/app/modules/tiers/services/sales.reclamation.problem.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesReclamationProblem } from '../../miscellaneous/model/SalesReclamationProblem';

@Injectable({
  providedIn: 'root'
})
export class SalesReclamationProblemService extends AppRestService<SalesReclamationProblem>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getReclamationProblems() {
    return this.getList(new HttpParams(), "sales-reclamation-problems");
  }

   addOrUpdateReclamationProblem(salesReclamationProblem: SalesReclamationProblem) {
    return this.addOrUpdate(new HttpParams(), "sales-reclamation-problem", salesReclamationProblem, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/SalesReclamationProblemService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.SalesReclamationProblem;

public interface SalesReclamationProblemService {

        public List<SalesReclamationProblem> getReclamationProblems();

        public SalesReclamationProblem addOrUpdateReclamationProblem(SalesReclamationProblem salesReclamationProblem);
}
